browse-next/src/views/DeviceDiagnosticsSubView.vue

```diff
@@ -46,11 +63,137 @@ const deviceConfig = ref<LoadedResource<DeviceConfigDetail>>(null);
 const currentLocationForDevice = ref<LoadedResource<ApiLocationResponse>>(null);
 const lastPowerOffTime = ref<LoadedResource<Date>>(null);
 const lastPowerOnTime = ref<LoadedResource<Date>>(null);
+// Device Settings
+const settings = ref<ApiDeviceHistorySettings | null>(null);
+const showCustomModal = ref(false);
+const customSettings = ref<Omit<WindowsSettings, "updated">>({
+  powerOff: "+30m",
+  powerOn: "-30m",
+  startRecording: "-30m",
+  stopRecording: "+30m",
+});
+
+const saltNodeGroup = ref<LoadedResource<string>>(null);
 const isLoading = (val: Ref<LoadedResource<unknown>>) =>
   computed<boolean>(() => val.value === null);
 const configInfoLoading = isLoading(deviceConfig);
 const versionInfoLoading = isLoading(versionInfo);
 const locationInfoLoading = isLoading(currentLocationForDevice);
+const nodeGroupInfoLoading = isLoading(saltNodeGroup);
+const settingsLoading = isLoading(settings);
+
+const lastUpdateWasUnsuccessful = ref<boolean>(true);
+
+const fields = [
+  { key: "name", label: "Setting" },
+  { key: "value", label: "Value" },
+  { key: "actions", label: "Actions" },
+];
+
+const currentWindowsType = computed(() => {
+  if (!settings.value || !settings.value.windows) {
+    return "unknown";
+  }
+  const { powerOn, powerOff, startRecording, stopRecording } =
+    settings.value.windows;
+  if (
+    powerOn === "-30m" &&
+    powerOff === "+30m" &&
+    startRecording === "-30m" &&
+    stopRecording === "+30m"
+  ) {
+    return "default";
+  } else if (
+    powerOn === "12:00" &&
+    powerOff === "12:00" &&
+    startRecording === "12:00" &&
+    stopRecording === "12:00"
+  ) {
+    return "24hour";
+  } else {
+    return "custom";
+  }
+});
+
+const formatTime = (timeString: string) => {
+  if (timeString[0] === "+" || timeString[0] === "-") return timeString;
+  const [hours, minutes] = timeString.split(":");
+  return `${hours}:${minutes}`;
+};
+
+const formatRecordingWindows = (windows: WindowsSettings) => {
+  return `Power On: ${formatTime(windows.powerOn)}, Power Off: ${formatTime(
+    windows.powerOff
+  )}, Start Recording: ${formatTime(
+    windows.startRecording
+  )}, Stop Recording: ${formatTime(windows.stopRecording)}`;
+};
+
+const settingsTable = computed(() => {
+  const rows = [
+    {
+      name: "Use Low Power Mode",
+      value: settings.value?.thermalRecording?.useLowPowerMode ?? false,
+    },
+    {
+      name: "Recording Windows",
+      value: settings.value?.windows
+        ? formatRecordingWindows(settings.value.windows)
+        : "Not set",
+    },
+  ];
+
+  return rows;
+});
+
+const fetchSettings = async () => {
+  try {
+    const response = await getSettingsForDevice(deviceId);
+    if (response.success) {
+      return response.result.settings;
+    }
+  } catch (e) {
+    console.error(e);
+  }
+  return null;
+};
+
+const handleToggleUseLowPowerMode = async (setting: { name: string }) => {
```

Comment:
With all of these toggles, there needs to be some kind of "change in progress" variable that disables the button while the change is happening, otherwise users will be able to mash the buttons while the request is in flight.